make/devices.cmake

```diff
@@ -1047,13 +1047,13 @@ function(ADD_FPGA_TARGET)
     add_custom_command(
       OUTPUT ${OUT_BIN}
       COMMAND ${BIT_TO_BIN_CMD_FOR_TARGET_LIST}
-      DEPENDS ${BIT_TO_BIN} ${OUT_BITSTREAM}
+      DEPENDS ${OUT_BITSTREAM}
       )
 
     add_custom_target(${NAME}_bin DEPENDS ${OUT_BIN})
     add_output_to_fpga_target(${NAME} BIN ${OUT_LOCAL_REL}/${TOP}.bin)
 
-    get_target_property_required(PROG_TOOL ${BOARD} PROG_TOOL)
+    get_target_property(PROG_TOOL ${BOARD} PROG_TOOL)
```

Comment:
This should be required.

Reply:
I hit issues with some PROG_TOOLS being defined as a *_TARGET. When those targets expanded to empty string, it would have PROG_TOOL expand to empty string (and error).

Specifically I think it was just two definitions that were causing the issue:

https://github.com/SymbiFlow/symbiflow-arch-defs/blob/master/ice40/boards.cmake#L51
https://github.com/SymbiFlow/symbiflow-arch-defs/blob/master/ice40/boards.cmake#L69

Should they be changed to using their `${TINYFPGAB}` program name instead? Except if changing the `TINYFPGAB` to a full path that might be annoying?

Reply:
I guess you are right, they should be targets, so not required.